File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

int visited[202];

void dfs(int idx, vector<vector<int>> computers) {
    for (int i=0; i<computers.size(); i++){
        if(i == idx || computers[idx][i] == 0 || visited[i])continue;
        visited[i] = 1;
        dfs(i,computers);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for (int i=0; i<n; i++){
        if(visited[i] == 1)continue;
        else{
            answer++;
            visited[i] = 1;
            dfs(i,computers);
        }
    }
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static int[] visited = new int[202];
    static int ret =0;
    
    public void dfs(int idx, int[][] computers){
        for (int i=0; i< computers[idx].length; i++){
            if(computers[idx][i] == 1 && visited[i] == 0){
                visited[i] = 1;
                dfs(i,computers);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        for (int i=0; i<computers.length; i++){
            if (visited[i] == 0){
                visited[i] = 1;
                ret++;
                dfs(i,computers);
            }
        }
        answer = ret;
        return answer;
    }
}